lib/cmd/fh3/version.js

```diff
@@ -0,0 +1,153 @@
+/* globals i18n */
+var dayInMs = 86400000;
+var daysToCache = 1;
+var timeout = 2000;
+var PATTERN = /\+BUILD-NUMBER/;
+
+var fhc = require('../../fhc'),
+  ini = require('../../utils/ini'),
+  _ = require('underscore'),
+  request = require('request'),
+  semver = require('semver'),
+  util = require('util');
+
+module.exports = {
+  'desc': i18n._('Version info about the RHMAP instance we\'re connected to'),
+  'examples': [{
+    cmd: 'fhc version',
+    desc: i18n._('Displays version of FHC and RHMAP')
+  }],
+  'demand': [],
+  'alias': {
+    'json': 'j'
+  },
+  'describe': {
+    'json' : i18n._('Output into json format')
+  },
+  'url': 'sys/info/version',
+  'preCmd': function(params, cb) {
+    this.checkFHCUpToDate(function() {
+      return cb(null, params);
+    });
+  },
+  'postCmd': function(params, response, cb) {
+    if (!params.json) {
+      return cb(null, createTxtMessage(response, ini.get('fhclatest')));
+    }
+    return cb(null, createJsonMessage(response));
+  },
+  checkFHCUpToDate: function(cb) {
+    var cachedLatest = ini.get('fhclatest');
```

Comment:
In my personal opinion caching this doesn't make sense. Original intention of the author was to enable this command while being offline, but I think that we can just skip that information if request to npm failed.